Docs/CourseFormat.txt

```diff
@@ -86,13 +86,149 @@ Complete Course Template
 -- the player gains from successfully completing a course in Oni mode.
 
+
+
+#SONGSELECT:GROUP=Stepmania 5:BPMRANGE=120..160:Difficulty=Medium:MODS=2x;
+-- #SONGSELECT is a new alternative to the #SONG tag, which will (hopefully)
+-- be easier to use and understand. Instead of relying on values input in a
+-- very specific and poorly-documented order, this tag is composed of 
+-- a series of PARAM=VALUE statements, separated by :'s. 
+
+-- The following parameters are available for selecting a song:
+-- TITLE, GROUP, ARTIST, GENRE, DIFFICULTY, METER, BPMRANGE, DURATION, SORT
+-- With the exception of SORT, these can be used in any combination and order.
+-- (See the SORT parameter description below for a better explanation)
+-- If none are specified, then a song and difficulty will be chosen completely at random.
+-- If a combination is specified that results in no possible songs to select,
+-- then that #SONGSELECT entry is skipped.
+
+-- Parameter Desciptions:
+--  TITLE= A list of one or more song titles, separated by commas.
+--  Notes: 
+--  - The "title" here can be either the name of the song directory,
+--    the #TITLE, or the #TITLETRANSLIT.
+--  - Titles must be an *exact match*.
+--  - Unlike the #SONG tag, this will not take into account group names.
+--    If you want to specify a specific song from a specific group, use
+--    this along with the GROUP parameter.
+--  - This does not support names that have commmas.
+--  Example: 
+#SONGSELECT:TITLE=Goin' Under,Springtime;  
+
+--  GROUP= A list of one or more song groups, separated by commas.
+--  Notes:
+--  - The "group" refers to the directory name of the song group.
+--  - This does not support folders that have commas in their name.
+--  - Group names must be an *exact match*.
+--  Example: 
+#SONGSELECT:GROUP=Stepmania 5,In The Groove;
+  
+--  ARTIST= A list of one or more artist names, separated by commas.
+--  Notes:
+--  - The "artist" refers to either the #ARTIST or #ARTISTTRANSLIT value
+--    of the simfile.
+--  - This does not support artist names that contain commas.
+-- Example:
+#SONGSELECT:ARTIST=Kommisar,NegaRen;
+
+--  GENRE= A list of one or more song genres, separated by commas.
+--  Notes:
+--  - The "genre" refers to the #GENRE tag of a song.
+--  - Most songs don't actually have a #GENRE defined, which unfortunately
+--    makes this parameter not terribly useful at this time.
+-- Example:
+#SONGSELECT:GENRE=J-Pop,Black Metal;
+
+--  DIFFICULTY= A list of one or more difficulties, separated by commas.
+--  Notes:
+--  - There are a number of acceptable values for this, but for the sake
+--    of consistency, please try to use:
+--    Beginner,Easy,Medium,Hard,Challenge,Edit
+--    (if you really want to use whacky value names, you can find the rest
+--     defined in Diffiulty.cpp)
+--  Example:
+#SONGSELECT:DIFFICULTY=Medium,Hard;
+
+--  METER= A minimum and maximum value of a song's difficulty meter,
+--  separated by EXACTLY TWO (2) periods.
```

Comment:
Thoughts on using a dash? 8-12? Two periods is also fine, but was curious.

Reply:
The two periods was just staying in line with how `#SONG` handled it, but using a dash is probably the more intuitive choise (I even messed this up last night while I was writing some courses to test this out, and kept putting dashes instead of `..`)

Reply:
Ahhhh I didn't realize that's how #song did it. I wonder if just supporting both is feasible. Dash seems way more intuitive.

Reply:
I'm don't think we need to support both, I'm just gonna go with dash.